docs/src/sensitivity-analysis-svm.md

```diff
@@ -0,0 +1,240 @@
+import Random
+using Test
+import SCS
+import Plots
+using DiffOpt
+using LinearAlgebra
+import MathOptInterface
+const MOI = MathOptInterface
+nothing # hide
+```
+
+Construct separatable, non-trivial data points.
+```@example 1
+N = 100
+D = 2
+Random.seed!(6)
+X = vcat(randn(N, D), randn(N,D) .+ [4.0,1.5]')
+y = append!(ones(N), -ones(N))
+nothing # hide
+```
+
+Let's define the variables.
+```@example 1
+model = diff_optimizer(SCS.Optimizer) 
+MOI.set(model, MOI.Silent(), true)
+
+# add variables
+l = MOI.add_variables(model, N)
+w = MOI.add_variables(model, D)
+b = MOI.add_variable(model)
+nothing # hide
+```
+
+Add the constraints.
+```@example 1
+MOI.add_constraint(
+    model,
+    MOI.VectorAffineFunction(
+        MOI.VectorAffineTerm.(1:N, MOI.ScalarAffineTerm.(1.0, l)), zeros(N),
+    ), 
+    MOI.Nonnegatives(N),
+)
+
+# define the whole matrix Ax, it'll be easier then
+# refer https://discourse.julialang.org/t/solve-minimization-problem-where-constraint-is-the-system-of-linear-inequation-with-mathoptinterface-efficiently/23571/4
+Ax = Matrix{MOI.ScalarAffineTerm{Float64}}(undef, N, D+2)
+for i in 1:N
+    Ax[i, :] = MOI.ScalarAffineTerm.([1.0; y[i]*X[i,:]; y[i]], [l[i]; w; b])
+end
+terms = MOI.VectorAffineTerm.(1:N, Ax)
+f = MOI.VectorAffineFunction(
+    vec(terms),
+    -ones(N),
+)
+MOI.add_constraint(
+    model,
+    f,
+    MOI.Nonnegatives(N),
+)
+nothing # hide
+```
+
+Define the linear objective function and solve the SVM model.
+```@example 1
+objective_function = MOI.ScalarAffineFunction(
+                        MOI.ScalarAffineTerm.(ones(N), l),
+                        0.0,
+                    )
+MOI.set(model, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(), objective_function)
+MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+
+MOI.optimize!(model)
+
+loss = MOI.get(model, MOI.ObjectiveValue())
+wv = MOI.get(model, MOI.VariablePrimal(), w)
+bv = MOI.get(model, MOI.VariablePrimal(), b)
+nothing # hide
+```
+
+We can visualize the separating hyperplane. 
+
+```@example 1
+# build SVM points
+svm_x = [0.0, 3.0]
+svm_y = (-bv .- wv[1] * svm_x )/wv[2]
+
+p = Plots.scatter(X[:,1], X[:,2], color = [yi > 0 ? :red : :blue for yi in y], label = "")
+Plots.yaxis!(p, (-2, 4.5))
+Plots.plot!(p, svm_x, svm_y, label = "loss = $(round(loss, digits=2))", width=3)
+Plots.savefig("svm_separating.svg")
+nothing # hide
+```
+
+![svg](svm_separating.svg)
+
+# Experiments
+Now that we've solved the SVM, we can compute the sensitivity of program variables -- the separating hyperplane in our case -- with respect to perturbations in problem data -- the data points -- using DiffOpt. For illustration, we've explored two questions:
+
+- How does a change in labels of the data points (`y=1` to `y=-1`, and vice versa) affects the position of the hyperplane? This is achieved by finding gradient of `w`, `b` with respect to `y[i]`, classification label of the ith data point (do not confuse `y[i]` with the array of dual variables.)
```

Comment:
```suggestion
- How does a change in labels of the data points (`y=1` to `y=-1`, and vice versa) affect the position of the hyperplane? This is achieved by finding the gradient of `w`, `b` with respect to `y[i]`, the classification label of the ith data point.
```

Reply:
This one is tricky to justify with DiffOpt: there is no notion of an infinitesimal perturbation for the labels, I would be for removing it

Reply:
yep. i didn't have any clue about perturbation in categorical data, but the change in value of the label (although not smooth) affects the hyperplane position